core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -596,6 +599,26 @@ object SparkSubmit extends CommandLineUtils {
       }
     }
 
+    if (isKubernetesCluster) {
```

Comment:
Hey Guys, really excited about this work by the way. :)

Just wondering if client mode from within a kubernetes cluster will be supported? Not looking to add work, just curious.

I'm wondering whether we could check if the client IP is within the kubernetes container CIDR (this info should be available via the kubernetes API), instead of just blocking client mode altogether. This would support Zeppelin/Jupyter instances  running within kube connecting to spark. Which is a big usecase in our case. 

We could take the [nodelist](https://kubernetes.io/docs/api-reference/v1/definitions/#_v1_nodelist) and check if the client IP is in any of the [v1.NodeSpec.PodCIDR](https://kubernetes.io/docs/api-reference/v1/definitions/#_v1_nodespec) of any node. Although as k8s clusters get large (nodes>1000), this may become a costly operation.

This would allow for client mode support, provided the client is on one of the kube nodes.